File: src/main/java/banane/io/pdb/web/UserController.java
package banane.io.pdb.web;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.ui.Model;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import banane.io.pdb.model.User;
import banane.io.pdb.service.UserService;
import banane.io.pdb.validator.UserValidator;

@RestController
@RequestMapping("/api")
public class UserController {

    @Autowired
    private UserService userService;

    @Autowired
    private UserValidator userValidator;

    @PostMapping(value = "/registration")
    public User registration(@RequestBody User userForm, BindingResult bindingResult, Model model) {
        userValidator.validate(userForm, bindingResult);

        if (bindingResult.hasErrors()) {
            return userForm;
        }

        return userService.save(userForm);
    }

}

File: src/main/java/banane/io/pdb/web/UserController.kt
package banane.io.pdb.web

import banane.io.pdb.model.User
import banane.io.pdb.service.UserService
import banane.io.pdb.validator.UserValidator
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.ui.Model
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class UserController {
    @Autowired
    private val userService: UserService? = null
    @Autowired
    private val userValidator: UserValidator? = null

    @PostMapping(value = ["/registration"])
    fun registration(@RequestBody userForm: User, bindingResult: BindingResult, model: Model?): User? {
        userValidator!!.validate(userForm, bindingResult)
        return if (bindingResult.hasErrors()) {
            userForm
        } else userService!!.save(userForm)
    }
}